### ISSUE TITLE
`FindBoost.cmake` warnings

### ISSUE THREAD

**Issue by marcfehling** [dealii/dealii#13106]
With `boost 1.78.0` and `cmake 3.22.1`, I get warnings in the `FindBoost.cmake` stage. Warnings might also occur in previous releases.

```
CMake Warning at /usr/share/cmake/Modules/FindBoost.cmake:1369 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindBoost.cmake:1492 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake/Modules/FindBoost.cmake:2102 (_Boost_MISSING_DEPENDENCIES)
  cmake/macros/macro_find_package.cmake:27 (_FIND_PACKAGE)
  cmake/modules/FindBOOST.cmake:60 (FIND_PACKAGE)
  cmake/macros/macro_find_package.cmake:27 (_FIND_PACKAGE)
  cmake/configure/configure_2_boost.cmake:144 (FIND_PACKAGE)
  build/CMakeFiles/CMakeTmp/evaluate_expression.tmp:1 (FEATURE_BOOST_FIND_EXTERNAL)
  cmake/macros/macro_evaluate_expression.cmake:30 (INCLUDE)
  cmake/macros/macro_configure_feature.cmake:237 (EVALUATE_EXPRESSION)
  cmake/configure/configure_2_boost.cmake:280 (CONFIGURE_FEATURE)
  cmake/macros/macro_verbose_include.cmake:19 (INCLUDE)
  CMakeLists.txt:121 (VERBOSE_INCLUDE)
```

**Comment by Rombur** [comment]
You get that warning because the version of boost you are using is newer than the version of cmake. Boost does not promise to the dependencies between their library to be stable between two releases. This means that cmake only guarantees to work with boost version that existed when cmake was released. 

Every time there is a new version of Boost or if you use a version of cmake that was released before the version of Boost you want to use, you get that warning. It has been like that for years.

**Comment by marcfehling** [comment]
Thanks for clarifying! `cmake 3.22.1` has been released on Dec 7th, `boost 1.78.0` on Dec 8th :-)
I've never seen this warning before, so thought our configuration is out-of-date. This is not the case then.
